src/components/structures/TimelinePanel.js

```diff
@@ -466,6 +470,21 @@ var TimelinePanel = React.createClass({
         this._reloadEvents();
     },
 
+    onAccountData: function(ev, room) {
+        if (this.unmounted) return;
+
+        // ignore events for other rooms
+        if (room !== this.props.timelineSet.room) return;
+
+        if (ev.getType() !== "m.fully_read") return;
+
+        const markerEventId = ev.getContent().event_id;
+        console.log('TimelinePanel: Read marker received from server', markerEventId);
+
+        this.setState({
```

Comment:
it seems like a problem that we aren't updating `roomReadMarkerTsMap` here.

I suspect we actually no longer need roomReadMarkerTsMap, but if that's true, let's get rid of it (preferably in a separate PR) rather than leave it in an inconsistent state.